src/Configuration/Dotenv.hs

```diff
@@ -1,86 +1,66 @@
 -- |
 -- Module      :  Configuration.Dotenv
--- Copyright   :  © 2015–2016 Stack Builders Inc.
+-- Copyright   :  © 2015–2017 Stack Builders Inc.
 -- License     :  MIT
 --
 -- Maintainer  :  Stack Builders <hackage@stackbuilders.com>
 -- Stability   :  experimental
 -- Portability :  portable
 --
--- This module contains common functions to load and read dotenv files.
+-- Provides the ability to load a dotenv file with an specific configuration.
 
-{-# LANGUAGE CPP #-}
+{-# LANGUAGE CPP             #-}
+{-# LANGUAGE RecordWildCards #-}
 
 module Configuration.Dotenv
-  ( load
```

Comment:
I'm concerned with backward compatibility here. I think we should not break it like this. I know a number of projects now that rely on the functions you deleted. And for some functions you do not provide a sensible and handy replacement. I remember adding `onMissingFile` and it's very useful.

More generally, I do think we should add the new functionality so it coexists with existing functionality, not replace it. I'd like my app to error early if a variable is not present in some cases, not use some default value. In some cases there is no good default.